src/main/java/edu/umich/ctools/sectionsUtilityTool/Utils.java

```diff
@@ -182,4 +193,56 @@ private static void logMsg(String uniqname, String originalUrl, HttpServletReque
 		M_log.info(String.format(baseString, user, originalUrl));
 	}
 
+	public static String makeApiCall(String... urlChunks){
+		String url = canvasURL + CANVAS_API_VERSION;
+		for (String chunk : urlChunks) {
+			url = url + chunk;
+		}
+		M_log.info("API call: " + url);
+		HttpResponse response = Utils.processApiCall(new HttpGet(url));
+		if (response == null) {
+			M_log.error(String.format("The Api call %s failed with errors", url));
+			return null;
+		}
+		int statusCode = response.getStatusLine().getStatusCode();
+		if (statusCode != HttpStatus.SC_OK) {
+			M_log.error(String.format("The Api call %s has failed with status code %s", url, statusCode));
+			return null;
+		}
+		String apiResponse;
+		String errMsg = "The API call %s has failed to extract the response due to %s";
+		try {
+			apiResponse = EntityUtils.toString(response.getEntity(), "UTF-8");
+		} catch (IOException e) {
+			M_log.error(String.format(errMsg, url, e.getMessage()));
+			return null;
+		} catch (Exception e) {
+			M_log.error(String.format(errMsg, url, e.getMessage()));
+			return null;
+		}
+
+		return apiResponse;
+	}
+
+	public static HttpResponse processApiCall(HttpUriRequest clientRequest) {
+		HttpClient client = HttpClients.createDefault();
+		final ArrayList<NameValuePair> nameValues = new ArrayList<NameValuePair>();
+		nameValues.add(new BasicNameValuePair("Authorization", "Bearer" + " " + canvasToken));
+		nameValues.add(new BasicNameValuePair("content-type", "application/json"));
+		for (final NameValuePair h : nameValues) {
+			clientRequest.addHeader(h.getName(), h.getValue());
+		}
+		HttpResponse response = null;
+		long startTime = System.currentTimeMillis();
+		try {
+			response = client.execute(clientRequest);
+		} catch (IOException e) {
+			M_log.error("Canvas API call did not complete successfully" + e.getMessage());
+		}
+		long stopTime = System.currentTimeMillis();
```

Comment:
The timing calculation and print should be in a finally clause.  That means the timing will always be printed and that can be useful if there is an exception that isn't an IOException.